source/DurationPicker.js

```diff
@@ -0,0 +1,781 @@
+	* @property {String} name - contains the name of this control.
+	* @public
+	*/
+
+	/**
+	* `moon.DurationPicker` is a [control]{@link enyo.Control} that can display -- or allow the
+	* selection of -- a duration expressed in hours, minutes and seconds, the selection
+	* display is configurable using template.
+	*
+	* @class moon.DurationPicker
+	* @ui
+	* @public
+	*/
+
+	enyo.kind(
+		/** @lends moon.DurationPicker.prototype */ {
+
+		/**
+		* @private
+		*/
+		name: 'moon.DurationPicker',
+
+		/**
+		* @private
+		*/
+		kind: 'moon.ExpandableListItem',
+
+		/**
+		* @private
+		*/
+		defaultKind: 'enyo.Control',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-expandable-picker',
+
+		/**
+		* @private
+		*/
+		handlers: {
+			//* Handler for _onChange_ events coming from constituent controls
+			onChange: 'handleChangeEvent'
+		},
+
+		events: {
+			/**
+			* {@link moon.DurationPicker#event:onDurationChange}
+			*/
+			onDurationChange: '',
+
+			/**
+			* {@link moon.DurationPicker#event:onCountdownExpired}
+			*/
+			onPickerExpired: ''
+		},
+
+		/**
+		* @public
+		* @lends moon.DurationPicker.prototype
+		*/
+		published: {
+
+			/**
+			* Text to be displayed as the current value if no item is currently selected
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			noneText: '',
+
+			/**
+			* The value of the picker, expressed as a standard JavaScript 
+			* object.
+			*
+			* @type {String}
+			* @default null
+			* @public
+			*/
+			value: null,
+
+			/**
+			* Optional label for hour
+			*
+			* @type {String}
+			* @default moon.$L('hour')
+			* @public
+			*/
+			hourText: moon.$L('hour'),		// i18n 'HOUR' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for minute
+			*
+			* @type {String}
+			* @default moon.$L('minute')
+			* @public
+			*/
+			minuteText: moon.$L('minute'),		// i18n 'MINUTE' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for second
+			*
+			* @type {String}
+			* @default moon.$L('second')
+			* @public
+			*/
+			secondText: moon.$L('second'),		// i18n 'SECOND' label in moon.DurationPicker widget
+
+			/**
+			* The template format of the picker
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			template: '',
+
+			/**
+			* Enable and disable the timer feature of duration picker
+			*
+			* @type {Boolean}
+			* @default true
+			* @public
+			*/
+			enableTimer: true
+		},
+
+		/**
+		* @type {Boolean}
+		* @default false
+		* @private
+		*/
+		countdown: false,
+
+		/**
+		* @type {String}
+		* @default null
+		* @private
+		*/
+		timer: null,
+
+		/**
+		* @type {Object}
+		* @default {}
+		* @private
+		*/
+		values: {},
+
+		/**
+		* @type {Array}
+		* @default []
+		* @private
+		*/
+		templateArray: [],
+
+		/**
+		* Optional label for Start Button
+		*
+		* @type {String}
+		* @default moon.$L('Start')
+		* @private
+		*/
+		startText: moon.$L('Start'),			// i18n 'Start' label in moon.DurationPicker buttons
```

Comment:
Having these as properties on the kind definitely works - I was curious if you had tried @Djspaceg's approach as a more efficient alternative: https://github.com/enyojs/moonstone/pull/1978#discussion_r25743288.